src/AppBundle/Entity/Adherent.php

```diff
@@ -588,34 +589,50 @@ public function setProcurationManagedAreaCodesAsString(string $codes = null): vo
         $this->procurationManagedArea->setCodesAsString((string) $codes);
     }
 
-    /**
-     * @return CommitteeMembership[]|Collection
-     */
-    public function getMemberships()
+    final public function getMemberships(): CommitteeMembershipCollection
     {
+        if (null === $this->memberships) {
```

Comment:
This code looks broken as it would decorate the collection on each call, try instead:
```php
if ($this->memberships instanceof Collection) {
    if (!$this->memberships instanceof CommitteeMembershipCollection) {
        $this->memberships = new CommitteeMembershipCollection($this->memberships->toArray());
    }
} else {
    $this->memberships = new CommitteeMembershipCollection((array) $this->memberships);
}

return $this->memberships;
```